src/libsystemd-network/dhcp-internal.h

```diff
@@ -75,6 +75,12 @@ int dhcp_packet_verify_headers(DHCPPacket *packet, size_t len, bool checksum, ui
 
 void dhcp_client_set_test_mode(sd_dhcp_client *client, bool test_mode);
 
+int dhcp_client_set_state_callback(
+                sd_dhcp_client *client,
+                sd_dhcp_client_callback_t cb,
+                void *userdata);
+int dhcp_client_get_state(sd_dhcp_client *client);
```

Comment:
Hm? The prototypes are duplicated.

Reply:
I was unsure what to do here, so I tried to follow the pattern that appeared to be established by `dhcp-internal.h` and `dhcp-client-internal.h` where `dhcp-client-internal.h` duplicates and exposes only the internal interfaces required by the dhcp client.

However, looking now more closely it seems this isn't used consistently. For example `networkd-dhcp4.c` includes `dhcp-client-internal.h` while `networkd-dhcp-common.c` includes `dhcp-internal.h` but only uses `dhcp_option_hash_ops` exposed by `dhcp-client-internal.h`.

For me it seems the right thing is to define the interfaces needed by the client only once in `dhcp-client-internal.h` and include that in `dhcp-internal.h` and replace the use of `dhcp-internal.h` with `dhcp-client-internal.h` within `networkd-dhcp-common.c`, and do the same for dhcp6. I'll try this in my next update and see if you prefer it.

Reply:
Yeah, I am always confused by the current messed state of the headers.
I really want to clean up them. If you want, please do in this PR or another later PR.

I requested here is only that please drop the duplicated prototypes.

If there already exist some duplicated prototypes, please also drop them if you'd like.

Reply:
I have removed the duplicated prototypes and changed `networkd-dhcp-common.c` to use `dhcp-client-internal.h`.